spec/site/tlc/alarm_spec.rb

```diff
@@ -20,7 +20,66 @@
   # the test is recommended.
 
   describe 'Alarm' do
-    before { skip "Don't yet have a reliable way of triggering alarms" }
+        
+    # Validate that a detector logic fault A0302 raises and removed.
+    # 1. Given the site is connected
+    # 2. We trigger an alarm by setting S0003 input 152 to True
+    # 3. Then we should receive an active alarm
+    # 4. We remove an alarm by setting S0003 input 152 to False
+    # 5. Then we should receive an inactive alarm
+
+    specify 'A0301 is raised when S0003 input 152 is forced to True', :script, sxl: '>=1.0.7' do |example|
+      Validator::Site.connected do |task,supervisor,site|
+
+        # Alarm is raised by setting S0003 input 152 to True
+        prepare task, site
+        ensure_input('True', 152) do
+          site.log "Waiting for alarm", level: :test
+          start_time = Time.now
+          alarm_code_id = 'A0301'
+          collector = site.collect_alarms num: 1, component: component, aCId: alarm_code_id,
```

Comment:
i suspect collect_alarms() is not working. alarm tests have been disabled for a while, and i did a major refactoring of the Collector classes. i think we need a proper AlarmCollector class. i'll look at that

Reply:
did you try to run the test? does it collect the expected method?

Reply:
Perfect! I run the test. see https://github.com/rsmp-nordic/rsmp_validator/runs/5479066165?check_suite_focus=truehttps://github.com/rsmp-nordic/rsmp_validator/runs/5479066165?check_suite_focus=true

Reply:
well, it seems to pass, but the detailed log for that test shows no Alarm being received:
https://github.com/rsmp-nordic/rsmp_validator/runs/5479066165?check_suite_focus=truehttps://github.com/rsmp-nordic/rsmp_validator/runs/5479066165?check_suite_focus=true#step:5:147

if you run this against the local ruby TLC emulator, it should fail.

i think what's happening is this: most of the block is passed as blocks to ensure_input(), but that method doesn't use the block for anything and doesn't call it. so most of the code is never run.